src/calculator/service.ts

```diff
@@ -130,6 +156,28 @@ function calculateSocialCredibility(user: TwitterUser, maxFollowers: number) : n
   return followersImpactCalc + ffProportionCalc
 }
 
+async function socialCredibilityFF(userId: string) {
+  return getUserFollowingsIds(userId)
+    .then(followingsList => {
+      return getUserFollowersIds(userId)
+        .then(followersList => {
+          const ff = followersList.ids.filter(id => followingsList.ids.includes(id))
+          return Promise.all(ff.map(getUserInfo))
+            .then(ffUserInfo => {
+              return getUserInfo(userId)
+                .then(userInfo => {
+                  const sumFF = ffUserInfo.map(user => ffProportion(user.followersCount, user.friendsCount)).reduce((a, b) => a + b, 0)
+                  const sizeFF = ff.length
+                  const userFFProportion = ffProportion(userInfo.followersCount, userInfo.friendsCount)
+                  return {
+                    credibility: (100 * sumFF) / (sizeFF * userFFProportion)
+                  }
+                })
+            })
+        })
+    })
+}
```

Comment:
This is an antipattern, named n+1 queries. Let's count the number of petitions that we are making to the twitter api:

* 1 one line 160
* 1 one line 162
* n on line 165, where n = |FF|.

For a simple twitter user like me, that will execute on each request to our api 100 or more requests to twitter, where 2 or those 100 are to get the following and followers, so we rest 98 to get users. our limit for the get user info by id endpoint is 900 daily requests, so this solution is not going to scale. This only doesn't affect our api usage, but our response time.

There is a version of the followings and followers endpoint that returns `users` instead of `ids`, use that. That way we are only going to have 2 queries.